src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -451,7 +455,7 @@ exprt string_constraint_generatort::add_axioms_for_parse_int(
     for(unsigned j=1; j<size; j++)
     {
       mult_exprt radix_sum(sum, radix);
-      if(j>=9)
+      if(j>=max_string_length-1)
```

Comment:
Actually since the radix is not fixed, we don't now what size can cause an overflow, so this condition does not make sense any-more. We should add the following axiom at almost each iteration (we can determine what is the smallest string that can cause an overflow) .